app/src/types.ts

```diff
@@ -0,0 +1,81 @@
+import { Dispatch, SetStateAction } from 'react'
+import { UrlObject } from 'url'
+
+/*
+ * Types relating to persistent user data
+ */
+export type Clinic = {
+  id: number
+  agency: string
+  agencyAddress: string
+  agencyTelephone: string
+  clinic: string
+  clinicAddress: string
+  clinicTelephone: string
+  county: string
+  zip: string
+}
+
+export type ChooseClinicData = {
+  clinic: Clinic | undefined
+  zipCode: string
+}
+
+export type ContactData = {
+  firstName: string
+  lastName: string
+  phone: string
+  comments: string
+}
+
+export type EligibilityData = {
+  residential: string
+  categorical: string[]
+  previouslyEnrolled: string
+  adjunctive: string[]
+}
+
+export type IncomeData = {
+  householdSize: string
+}
+
+export type SessionData = {
+  chooseClinic: ChooseClinicData
+  contact: ContactData
+  eligibility: EligibilityData
+  income: IncomeData
+  submitted: boolean
+}
+
+export interface ReadSession {
+  session: SessionData
+}
+
+export interface WriteSession extends ReadSession {
+  setSession: Dispatch<SetStateAction<SessionData>> | (() => unknown)
+}
+
+export interface ClearSession extends WriteSession {
+  sessionKey: string
+}
```

Comment:
Notes for reviewers: These types represent data needed for different types of access "permissions" related to `SessionData`.